### ISSUE TITLE
Detection issue with custom regex

### ISSUE THREAD

**Issue by arbazkiraak** [zricethezav/gitleaks#164]
Hello @zricethezav 

I have added a regex to detect password in urls. But not sure why gitleaks is not able to detect it.

`https://raw.githubusercontent.com/hackerarbaz/Tryme/master/slakc.py`

````
→ gitleaks --version
1.22.0

→ gitleaks --repo=https://github.com/hackerarbaz/Tryme --config=/root/cimex2/gitleaks-back.toml -v
INFO[2018-12-14T04:51:35Z] cloning https://github.com/hackerarbaz/Tryme 
Enumerating objects: 57, done.
Counting objects: 100% (57/57), done.
Compressing objects: 100% (49/49), done.
Total 57 (delta 18), reused 0 (delta 0), pack-reused 0
INFO[2018-12-14T04:51:37Z] 0 leaks detected. 20 commits inspected in 2 seconds 347 milliseconds 


→ cat gitleaks-back.toml 
title = "gitleaks config"
[[regexes]]
description = "WOOT"
regex = '''[a-zA-Z]{3,10}://[^/\s:@]{3,20}:[^/\s:@]{3,20}@.{1,100}[\"'\s]'''

[whitelist]
regexes = [
  # "AKIA.*EXAMPLE",
]
files = [
  "(.*?)(jpg|gif|doc|pdf|bin|go)$"
]

**Comment by arbazkiraak** [comment]
Hello @milo-minderbinder ,

> Thanks alot for detailed info, i will be more careful from next time.

**Comment by arbazkiraak** [comment]
I had never used `grep` with any specific expression, Just wondering why it detects the pattern in `Perl Expression only`. @milo-minderbinder  I know it is out of scope topic, but would be helpful for me to understand it.

**Comment by milo-minderbinder** [comment]
If you're asking asking why `grep -P "[a-zA-Z]{3,10}://[^/\s:@]{3,20}:[^/\s:@]{3,20}@.{1,100}"` works but `grep "[a-zA-Z]{3,10}://[^/\s:@]{3,20}:[^/\s:@]{3,20}@.{1,100}"` does not, it's because the `-P` flag tells `grep` to interpret the regex as a Perl-compatible regular expression (a.k.a. PCRE). Perl's regular expressions provide more functionality than basic or extended regular expression syntax (which `grep` uses by default), and includes support for special escape characters to represent character classes (e.g. `\s` represents any whitespace character, whereas in basic or extended regex syntax, you'd have to use the posix `[[:space:]]` named set), named groups & references (e.g. `(?<quote>['"])foo\g{quote}`), and more.

**tl;dr:** `grep` won't work without the `-P` switch with your regular expression because the `\s` (which represents a whitespace character) is only supported in Perl-compatible regular expressions.

**Comment by arbazkiraak** [comment]
Awesome @milo-minderbinder  , Thanks for helping me understanding it.
